src/realm/object-store/shared_realm.hpp

```diff
@@ -395,6 +395,8 @@ class Realm : public std::enable_shared_from_this<Realm> {
     // Closing a Realm will wait for any asynchronous writes which have been commited but not synced
     // to sync. Asynchronous writes which have not yet started are canceled.
     void close();
+    // close realm and delete all the realm files
+    void close_and_delete_realm();
```

Comment:
Hi @nicola-cab Doing it this way with a public method will also help Kotlin. Right now we have a similar use case, but we need to run this logic _after_ the Realm is opened (which this seems to support) and we are running into issues deleting the file using existing API's as the SyncClient are keeping the Realm open.